src/vm/runners/cairo_runner.rs

```diff
@@ -297,6 +304,80 @@ impl CairoRunner {
         Ok(())
     }
 
+    pub fn get_memory_holes(&self, vm: &VirtualMachine) -> Result<usize, MemoryError> {
+        let accessed_addresses = vm
+            .accessed_addresses
+            .as_ref()
+            .ok_or(MemoryError::MissingAccessedAddresses)?;
+
+        let mut builtin_accessed_addresses = vm
+            .builtin_runners
+            .iter()
+            .flat_map(|(_, x)| x.get_memory_accesses(self).into_iter())
+            .collect::<HashSet<_>>();
+
+        builtin_accessed_addresses.extend(accessed_addresses.iter().cloned());
+        vm.segments.get_memory_holes(&builtin_accessed_addresses)
+    }
+
+    pub fn end_run(
+        &mut self,
+        _disable_trace_padding: Option<bool>,
+        disable_finalize_all: Option<bool>,
+        vm: &mut VirtualMachine,
+    ) -> Result<(), VirtualMachineError> {
+        // let disable_trace_padding = disable_trace_padding.unwrap_or(true);
+        let disable_finalize_all = disable_finalize_all.unwrap_or(false);
+
+        if self.run_ended {
+            return Err(RunnerError::RunAlreadyEnded.into());
+        }
+
+        // Process accessed_addresses.
+        {
+            let accessed_addresses = vm
+                .accessed_addresses
+                .as_ref()
+                .ok_or_else::<VirtualMachineError, _>(|| {
+                    MemoryError::MissingAccessedAddresses.into()
+                })?;
+            let mut new_accessed_addresses = Vec::with_capacity(accessed_addresses.len());
+
+            for addr in accessed_addresses {
+                let relocated_addr = vm
+                    .memory
+                    .relocate_value(Some(Cow::Owned(addr.into())))?
+                    .unwrap()
```

Comment:
Please remove this unwrap

Reply:
This unwrap will never be triggered because as long as the parameter to `relocate_value` is `Some(_)` it will return `Some(_)`.

What should I do in this case?

Reply:
Makes sense, then we should remove the Option from `Memory::relocate_value'`s input and output, as it serves no purpose